### PR TITLE
feat: integrate Streamlit

### PR DESCRIPTION
**Pull request by berrytern**
This pull request integrates Streamlit into the Langflow project, enhancing user interaction through key functionalities:

Chat Template: Added functionality to manage and use chat templates within Streamlit.
Send Chat Message: Implemented endpoints for sending chat messages directly through Streamlit.
Read Chat Messages: Added functionality to return chat messages.
Listen to Chat Messages: Introduced an endpoint to listen for incoming chat messages.

### CODE REVIEW

File: src/backend/base/langflow/streamlit/api/streamlit.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+user_input = st.chat_input("{model.input_msg}")
+
+if user_input:
+    with st.chat_message("user", avatar={user_avatar}):
+        st.markdown(user_input)
+    rq.post("http://localhost:7881/api/v1/sessions/"+st.session_state.session_id+"/messages", json=dict(role="user", content=user_input))
+
+sleep(2);st.rerun();
+
+"""
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/streamlit/api/streamlit.py:194]: This should be put in a separate file as a normal python file and them we can process it in a different way as well. Maybe setup reusable/editable secrets in the streamlit app that you reference in the file.

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
Hey @berrytern 

This is looking pretty good. We just need some changes to make it more maintainable. 

One point I'd make is to add the environment variables to the Settings: https://github.com/langflow-ai/langflow/blob/b7bc36d32ab85eda5efe2f9db3d46509c9d4220b/src/backend/base/langflow/services/settings/base.py#L60

and then using settings service to access them.

File: src/backend/base/langflow/__main__.py [hunk 2]
```diff
@@ -120,13 +120,19 @@ def run(
...
+        envvar="LANGFLOW_STREAMLIT_ENABLED",
+    ),
 ):
     """
     Run Langflow.
     """
 
     configure(log_level=log_level, log_file=log_file)
     set_var_for_macos_issue()
+    print("running", streamlit_enabled)
```
[the remark this answers is not in the document]
Comment (berrytern) [reply src/backend/base/langflow/__main__.py]: Thanks! I am now using logger.debug on Streamlit startup.

File: src/backend/base/langflow/main.py [hunk 3]
```diff
@@ -175,8 +202,17 @@ async def exception_handler(request: Request, exc: Exception):
                 content={"message": str(exc)},
             )
 
-    FastAPIInstrumentor.instrument_app(app)
+    if os.getenv("LANGFLOW_STREAMLIT_ENABLED", "false").lower() == "true":
+        from langflow.streamlit import StreamlitApplication
+        from .__main__ import wait_for_server_ready
```
[the remark this answers is not in the document]
Comment (berrytern) [reply src/backend/base/langflow/main.py:208]: now i am using the service settings

File: src/backend/base/langflow/streamlit/api/streamlit.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+            base_chat_data = loads(f.read())
+
+
+load_previous_chat()
+
+
+async def arun_flow(flow_id: str, api_key: str):
+    async with ClientSession() as session:
+        headers = {"x-api-key": api_key}
+        async with session.post(f"http://backend:7860/api/v1/run/{flow_id}", headers=headers) as r:
```
[the remark this answers is not in the document]
Comment (berrytern) [reply src/backend/base/langflow/streamlit/api/streamlit.py]: this part of code was not been used, i removed it

File: src/backend/base/langflow/main.py [hunk 5]
```diff
@@ -104,6 +104,33 @@ async def lifespan(app: FastAPI):
...
+    streamlit_app.include_router(streamlit_router, prefix="/api/v1")
+
+    @streamlit_app.get("/health")
+    async def health():
+        pass
+
+    config = uvicorn.Config(
+        streamlit_app,
+        host="0.0.0.0",
+        port=7881,
```
[the remark this answers is not in the document]
Comment (berrytern) [reply src/backend/base/langflow/main.py]: thanks! I tried to follow your advice.

**Pull request closed without merging**